packages/number-field/src/vaadin-number-field.js

```diff
@@ -215,9 +204,7 @@ export class NumberField extends InputFieldMixin(SlotStylesMixin(ThemableMixin(E
     super.connectedCallback();
 
     if (this.inputElement) {
-      this.inputElement.min = this.min;
-      this.inputElement.max = this.max;
-      this.__applyStep(this.step);
+      this.__setInputMode(this.inputElement, this.min);
```

Comment:
Wouldn't we also need to run this when the min constraint changes?

Reply:
Reverted this code per https://github.com/vaadin/web-components/pull/3071#discussion_r755370742